crates/egui/src/frame_state.rs

```diff
@@ -38,9 +38,12 @@ pub(crate) struct FrameState {
     /// Initialized to `None` at the start of each frame.
     pub(crate) tooltip_state: Option<TooltipFrameState>,
 
-    /// horizontal, vertical
+    /// The current scroll area should scroll to this range (horizontal, vertical).
     pub(crate) scroll_target: [Option<(Rangef, Option<Align>)>; 2],
 
+    /// The current scroll area should scroll by this much (horizontal, vertical).
+    pub(crate) scroll_delta: [Option<f32>; 2],
```

Comment:
This could just be a `Vec2`, no?

```suggestion
    pub(crate) scroll_delta: Vec2,
```

Reply:
Updated. I think we could also update scroll_target to be `Option<[(..); 2]>` instead of `[Option<(..)>; 2]`. That would make the code in Prepared::end a bit cleaner. Should I include that change as well?

Reply:
No - keep each PR as small as possible please